add_codeowners.py

```diff
@@ -0,0 +1,498 @@
+                          'sub-arch,sub-req,substitutions,tpl-arch,tpl-req}',
+}
+
+
+def create_codeowners_file(settings: RepoOwnerSettings) -> str:
+    """
+    Create codeowners file based on RepoOwnerSettings
+
+    See https://confluence.slac.stanford.edu/x/u5Q9Hw for details.
+    By default:
+    - "default" = sme_owners + area_owners
+    - .github owned by admins only
+    - Default (*): default
+    - language specific files: default + lang_owners
+
+    Returns content encoded as base64 utf-8, in order to ensure the
+    payload is appropriate for github's REST API
+    """
+    lines = []
+
+    # gather groups
+    default_owners = list(set(settings.sme_owners + settings.area_owners))
+    if not default_owners:
+        default_owners = [CodeownerGroup.ADMIN]
+    default_groups = ["@pcdshub/" + grp for grp in default_owners]
+
+    # .github admin group
+    lines.append("# github folder holds administrative files")
+    lines.append(f".github/** @pcdshub/{CodeownerGroup.ADMIN}")
+    lines.append("")
+
+    # default group
+    lines.append("# default group")
+    lines.append(f"* {' '.join(default_groups)}")
+    lines.append("")
+
+    # Language specific groups
+    if settings.lang_owners:
+        lines.append("# language-specific group(s)")
+        for group in settings.lang_owners:
+            specific_lang_group = ' '.join([f"@pcdshub/{group}"] + default_groups)
+            lines.append(f"{GROUP_TO_EXT[group]} {specific_lang_group}")
+        lines.append("")
+
+    base_file = "\n".join(lines)
+
+    # ensure file contents can be base64 encoded for submission to github API
+    encoded_content = base64.b64encode(base_file.encode('utf-8'))
+    return encoded_content.decode("ascii")
+
+
+def parse_repo_list(repo_data_path: str) -> List[RepoOwnerSettings]:
+    """
+    Read and parse a repository data list from ``repo_data_path``
+    Expects a basic csv format (comma-delimited, first row are column headers)
+    Expects the following header names
+    - 'owner': repository owner, organization
+    - 'repo_name': repository name
+    - 'sme_owners': subject matter experts
+    - 'area_owners': hutch/area owners
+    - 'lang_owners': programming language experts
+
+    Parameters
+    ----------
+    repo_data_path : str
+        path to the repo-data csv
+
+    Returns
+    -------
+    List[ProtectionGroup]
+        A list of ProtectionGroup's, holding settings for protection groups
+    """
+    data_path = Path(repo_data_path)
+    if not data_path.exists:
+        print('repo data file does not exist')
+        return
+    repo_data = []
+    # Deal with extra columns
+    with open(data_path, 'r') as csvfile:
+        csv_reader = csv.DictReader(csvfile)
+        for row in csv_reader:
+            data_row = RepoOwnerSettings.from_dict(row)
+            repo_data.append(data_row)
+    return repo_data
+
+
+@DryRunner()
+def create_fork(
```

Comment:
Question:
Is this still used? If so, is it better to keep it or remove it?

Reply:
It's not used at the moment, but I think there are worlds where we end up using it.  For public repos we can adjust this workflow quite easily to make a fork